src/Websites/Websites/Cmdlets/DeploymentSlots/SetAzureWebAppSlot.cs

```diff
@@ -133,6 +133,10 @@ public class SetAzureWebAppSlotCmdlet : WebAppSlotBaseCmdlet
         [ValidateSet("AllAllowed", "Disabled", "FtpsOnly")]
         public string FtpsState { get; set; }
 
+        [Parameter(ParameterSetName = ParameterSet1Name, Mandatory = false, HelpMessage = "Custom hostnames associated with web app")]
+        [ValidateNotNullOrEmpty]
+        public string[] HostNames { get; set; }
```

Comment:
plural names are discouraged from design perspective, please use "HostName" instead.

[parameter design guide](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/parameter-best-practices.md)

Reply:
@VeryEarly the current Set-AzWebApp has the parameter in plural so for consistency sake we would leave it the same https://docs.microsoft.com/en-us/powershell/module/Az.Websites/Set-AzWebApp?view=azps-3.4.0